gems/sorbet-runtime/lib/types/props/constructor.rb

```diff
@@ -3,18 +3,26 @@
 
 module T::Props::Constructor
   include T::Props::WeakConstructor
+end
 
-  def initialize(hash={})
-    decorator = self.class.decorator
+module T::Props::Constructor::DecoratorMethods
+  extend T::Sig
 
-    decorator.props.each do |prop, rules|
-      # It's important to explicitly compare against `true` here; the value can also be :existing or
-      # :on_load (which are truthy) but we don't want to treat those as optional in this context.
-      if T::Props::Utils.required_prop?(rules) && !decorator.has_default?(rules) && !hash.key?(prop)
-        raise ArgumentError.new("Missing required prop `#{prop}` for class `#{self.class}`")
+  # checked(:never) - O(runtime object construction)
+  sig {params(instance: T::Props::Constructor, hash: T::Hash[Symbol, T.untyped]).returns(Integer).checked(:never)}
+  def construct_props_without_defaults(instance, hash)
+    @props_without_defaults&.count do |p, setter_proc|
```

Comment:
what does `count` do here?
Can we make this code more explicit?

Reply:
Will add comment. (Thought I had one but must have disappeared in some refactoring.)